controllers/tasks.js

```diff
@@ -251,16 +281,33 @@ const getUserTasks = async (req, res) => {
  * @param req {Object} - Express request object
  * @param res {Object} - Express response object
  */
+
+/**
+ * @deprecated
+ * WARNING: This API endpoint is being deprecated and will be removed in future versions.
+ * Please use the updated API endpoint: `/tasks/:userId?dev=true` for retrieving user's task details.
+ *
+ * This API is kept temporarily for backward compatibility.
+ */
+
 const getSelfTasks = async (req, res) => {
   try {
     const { username } = req.userData;
 
     if (username) {
       if (req.query.completed) {
         const allCompletedTasks = await tasks.fetchUserCompletedTasks(username);
+        res.set(
+          "X-Deprecation-Warning",
+          "WARNING: This endpoint is deprecated and will be removed in the future. Please use /tasks/:userId to get the task details."
+        );
         return res.json(allCompletedTasks);
       } else {
         const allTasks = await tasks.fetchSelfTasks(username);
+        res.set(
+          "X-Deprecation-Warning",
+          "WARNING: This endpoint is deprecated and will be removed in the future. Please use /tasks/:userId to get the task details."
```

Comment:
Here in comment you mentioned, `tasks/:userId` as route, but below you update the route to `.../v1/:userId`?
Which is the correct one?

Reply:
my bad i will update them sir